src/sentry/integrations/slack/client.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.integrations.client import ApiClient
+from sentry.shared_integrations.exceptions import ApiError
+from sentry.utils import metrics
+
+SLACK_DATADOG_METRIC = "integrations.slack.http_response"
+
+
+class SlackClient(ApiClient):
+    allow_redirects = False
+    integration_name = "slack"
+    base_url = "https://slack.com/api"
+    datadog_prefix = "integrations.slack"
+
+    def __init__(self):
+        super(SlackClient, self).__init__()
+
+    def track_response_data(self, code, span, error=None, resp=None):
+        span.set_http_status(code)
+        span.set_tag("integration", "slack")
+
+        is_ok = False
+        # If Slack gives us back a 200 we still want to check the 'ok' param
+        if resp:
+            response = resp.json()
+            is_ok = response.get("ok")
+            span.set_tag("ok", is_ok)
+
+            # when 'ok' is False, we can add the error we get back as a tag
+            if not is_ok:
+                span.set_tag("slack_error", response.get("error"))
+
+        metrics.incr(
+            SLACK_DATADOG_METRIC,
+            sample_rate=1.0,
+            tags={"ok": False if is_ok is False else True, "status": code},
+        )
+
+        extra = {
+            self.integration_type: self.name,
+            "status_string": six.text_type(code),
+            "error": six.text_type(error)[:256] if error else None,
```

Comment:
@MeredithAnya If we get `ok=false`, I think we should use `response.get("error")` as the error here

Reply:
set the error to `response.get("error")` above